hosted-ce/30-remote-site-setup.sh

```diff
@@ -106,17 +120,27 @@ fi
 REMOTE_HOST_KEY=`ssh-keyscan -p "$remote_port" "$remote_fqdn"`
 [[ -n $REMOTE_HOST_KEY ]] || errexit "Failed to determine host key for $remote_fqdn:$remote_port"
 
-# HACK: Symlink the Bosco key to the location expected by
+
+users=$(get_mapped_users)
+[[ -n $users ]] || errexit "Did not find any user mappings"
+
+# Use the first user for things we only need once
+firstuser=$(printf "%s\n" "$users" | head -n1)
+id -u "$firstuser" &>/dev/null || errexit "Expected user $firstuser doesn't exist"
+
+firstuser_key=$(get_bosco_key "$firstuser")
+[[ -f $firstuser_key ]] || errexit "Failed to get SSH key for $firstuser"
+
+# HACK: Copy the Bosco key to the location expected by
 # bosco_cluster so it doesn't go and try to generate a new one
 root_ssh_dir=/root/.ssh/
 mkdir -p $root_ssh_dir
 chmod 700 $root_ssh_dir
-ln -s $BOSCO_KEY $root_ssh_dir/bosco_key.rsa
+install -o root -g root -m 0600 "$firstuser_key" $root_ssh_dir/bosco_key.rsa
 
```

Comment:
Can't we get rid of this if we're not using root for SSH anymore?

Reply:
I'd have to check what bosco_cluster does, to see if the hack mentioned in the comment is still needed. root is still the one running the ssh command, it's just using someone else's key.

Reply:
> I'd have to check what bosco_cluster does, to see if the hack mentioned in the comment is still needed.

I think we want to run `bosco_cluster` as the unpriv users so that it picks up each of their SSH configs/keys

> root is still the one running the ssh command, it's just using someone else's key.

Ah right